src/preact/base-element.js

```diff
@@ -130,6 +130,19 @@ export class PreactBaseElement extends AMP.BaseElement {
     }
   }
 
+  /**
+   * @protected
+   * @param {!JsonObject} props
+   */
+  mutateProps(props) {
+    if (this.defaultProps_) {
+      Object.assign(this.defaultProps_, props);
+    } else {
+      this.defaultProps_ = props;
```

Comment:
IMHO, this still has to be assign. E.g. the expression is simply:

```
this.defaultProps_ = {...this.defaultProps_, ...props}
```

Reply:
Dima's suggestion will work even if `defaultProps` is `null`/`undefiend` or an object.